src/components/Search/SearchPageHeader.tsx

```diff
@@ -44,10 +44,30 @@ type HeaderWrapperProps = Pick<HeaderWithBackButtonProps, 'icon' | 'children'> &
 
 function HeaderWrapper({icon, children, text, isCannedQuery}: HeaderWrapperProps) {
     const styles = useThemeStyles();
+    const [reports] = useOnyx(ONYXKEYS.COLLECTION.REPORT);
+    const taxRates = getAllTaxRates();
+    const [cardList = {}] = useOnyx(ONYXKEYS.CARD_LIST);
```

Comment:
this `HeaderWrapper` is a helper component that is **not** the main export from this file. I think if we want to keep it it needs to be extremely simple - basically a JSX wrapper for children.

Adding extra logic to it will make the whole `SearchPageHeader` very complex.
Refactor this, I have 2 general ideas in mind:
 - if we want to keep it just make it accept all arguments as react props. don't do useOnyx inside or any complex logic, pass it everything from the top, or...
 - do a bigger refactor of `SearchPageHeader` I think its already too complex. It accepts both `children` and `text` to display; it has several conditions inside to decide what is shown

I think it might be a good moment to clean this up